### PR TITLE
ADR 454 for new product pages

### PR DESCRIPTION
**Pull request by kr8n3r**
What
----

Document why and what we agreed for rebuilding the paas-product pages

How to review
-------------
[Rendered text](https://github.com/alphagov/paas-team-manual/blob/8ebca9645e1d7eaca324ac174c70f50ce330196c/source/architecture_decision_records/ADR454-new-product-pages-language-and-framework.html.md)

👀 and 🤔

Who can review
--------------
not me

### CODE REVIEW

File: source/architecture_decision_records/ADR454-new-product-pages-language-and-framework.html.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+With the above in mind we researched options. Our admin interface is built in React,
+so we narrowed the scope to React-based static site generators.
+
+We ended up comparing two: [NextJS] with static page export and [GatsbyJS] 
+(which exports static pages by default).
+For page content we agreed that writing pages in [Markdown] is a good option,
+so we tested both with [MDX] which can also embed React components inside content pages.
+
+[NextJS] and [GatsbyJS] have diffent approaches to development and there are minor
```
Comment (tlwr) [inline source/architecture_decision_records/ADR454-new-product-pages-language-and-framework.html.md:30]:
```suggestion
[NextJS] and [GatsbyJS] have different approaches to development and there are minor 
```

File: source/architecture_decision_records/ADR454-new-product-pages-language-and-framework.html.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+[PaaS product pages] have been reviewed and a number of accessibility issues
+have been identified. To resolve those we, would need to make an upgrade and
+and review if any additional changes are needed to align with the GOV.UK Design System.
+
+As those pages are built on Ruby on Rails and in [ADR024] we've made the decision
+to develop our user-facing applications on Node, it's a good opportunity to 
+look at rebuilding the product pages.
+
+We've disscussed user needs and it emerged that:
```
Comment (tlwr) [inline source/architecture_decision_records/ADR454-new-product-pages-language-and-framework.html.md:13]:
```suggestion
We've discussed user needs and it emerged that:
```

File: source/architecture_decision_records/ADR454-new-product-pages-language-and-framework.html.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
+# ADR454: new product pages language and framework
+
+## Context
+
+[PaaS product pages] have been reviewed and a number of accessibility issues
+have been identified. To resolve those we, would need to make an upgrade and
+and review if any additional changes are needed to align with the GOV.UK Design System.
+
+As those pages are built on Ruby on Rails and in [ADR024] we've made the decision
```
Comment (tlwr) [inline source/architecture_decision_records/ADR454-new-product-pages-language-and-framework.html.md:9]:
```suggestion
As those pages are built in Ruby and in [ADR024] we've made the decision
```

File: source/architecture_decision_records/ADR454-new-product-pages-language-and-framework.html.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+to develop our user-facing applications on Node, it's a good opportunity to 
+look at rebuilding the product pages.
+
+We've disscussed user needs and it emerged that:
+
+* anyone in the team (developer and non-developer) should be able to update pages 
+with less effort
+* pages should be performant for end users
```
Comment (tlwr) [inline source/architecture_decision_records/ADR454-new-product-pages-language-and-framework.html.md:17]:
```suggestion
* pages should be performant for end users
* pages should be rendered by the server
```

Might be redundant with the preceding statement "performant" but also explicit

File: source/architecture_decision_records/ADR454-new-product-pages-language-and-framework.html.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+## Status
+
+Accepted.
+
+## Consequences
+
+This implementation will allow us to:
+
+- iterate our content more quickly as non-developers will be able to make chances easily
+- keep our page updated with newer releases of GOV.UK Design System to continue to makem
```
Comment (tlwr) [inline source/architecture_decision_records/ADR454-new-product-pages-language-and-framework.html.md:50]:
```suggestion
- keep our page updated with newer releases of GOV.UK Design System to continue to make them
```

File: source/architecture_decision_records/ADR454-new-product-pages-language-and-framework.html.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+to develop our user-facing applications on Node, it's a good opportunity to 
+look at rebuilding the product pages.
+
+We've discussed user needs and it emerged that:
+
+* anyone in the team (developer and non-developer) should be able to update pages 
+with less effort
+* pages should be performant for end users
+* pages should be rendered by the server
+* keeping pages up to date with GOV.UK Design System releases should be quiker and easier
```
Comment (pauldougan) [inline source/architecture_decision_records/ADR454-new-product-pages-language-and-framework.html.md:19]:
```suggestion
* keeping pages up to date with GOV.UK Design System releases should be quicker and easier
```

File: source/architecture_decision_records/ADR454-new-product-pages-language-and-framework.html.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+We ended up comparing two: [NextJS] with static page export and [GatsbyJS] 
+which exports static pages by default.
+For page content we agreed that writing pages in [Markdown] is a good option,
+so we tested both with [MDX] which can also embed React components inside content pages.
+
+[NextJS] and [GatsbyJS] have different approaches to development and there are minor 
+performance differences between them.
+
+Our usecase for now is narrow enough, and with the primary need of anyone in the team
```
Comment (pauldougan) [inline source/architecture_decision_records/ADR454-new-product-pages-language-and-framework.html.md:34]:
```suggestion
Our use case for now is narrow enough, and with the primary need of anyone in the team
```

File: source/index.html.md [hunk 8]
```diff
@@ -130,3 +130,4 @@ This section contains Architecture Decision Records (ADR) as described in this b
  - [ADR451 BOSH access without SOCKS](architecture_decision_records/ADR451-bosh-access-without-socks/)
  - [ADR452 BOSH access with mTLS](architecture_decision_records/ADR452-bosh-access-with-mtls/)
  - [ADR453 Isolation segments](architecture_decision_records/ADR453-isolation-segments/)
+ - [ADR454: new product pages language and framework selection](architecture_decision_records/ADR454-new-product-pages-language-and-framework/)
```
Comment (AP-Hunt) [inline source/index.html.md:133]: No colon to be consistent with the others

**Comment by AP-Hunt** [comment]
It would be good to document the fact that installing Gatsby was difficult/impossible for those without admin access on their machine. That was a big part of our decision making,

**Comment by kr8n3r** [comment]
@AP-Hunt added that

**Pull request merged**